<launch>
  <node name="fusion_detector" pkg="fusion_detector" type="fusion_detector_node" output="screen" >

    <!-- setup camera topic & detect_topic & camera lidar calibration -->
    <!-- support max 4 cameras -->
    <param name="camCount" value="3"/>

    <!--camera 1-->
    <param name="camera1_topic" value="/camera_perception/perception/front/image_raw" />
    <param name="camera1_detect_results" value="/camera_perception/perception/front/detection_results" />
    <param name="camera1_cali_file" value="/opt/miivii/data/calibration/camera_lidar/perception_front.yaml" />

    <!--camera 2-->
    <param name="camera2_topic" value="/camera_perception/perception/left/image_raw" />
    <param name="camera2_detect_results" value="/camera_perception/perception/left/detection_results" />
    <param name="camera2_cali_file" value="/opt/miivii/data/calibration/camera_lidar/perception_left.yaml" />

    <!--camera 3-->
    <param name="camera3_topic" value="/camera_perception/perception/right/image_raw" />
    <param name="camera3_detect_results" value="/camera_perception/perception/right/detection_results" />
    <param name="camera3_cali_file" value="/opt/miivii/data/calibration/camera_lidar/perception_right.yaml" />

    <param name="lidar_topic" value="/lidar_points" />

    <param name="visualize_fusion" value="false" /><!-- param, show fusion results window -->

    <param name="pose_estimation" value="true"/>

    <param name="downsample_cloud" value="true"/><!-- param, to use downsample for point_clouds -->

    <param name="debug_time" value="true" /><!-- param, to show process time -->


    <!--             advertise topic                  -->
    <param name="bounding_boxes_topic" value="/fusion_detector/bounding_boxes" />

    <param name="combined_objects_labels" value="/fusion_detector/combined_objects_labels" />

    <!--publish point clouds without detect objects-->
    <param name="enbable_publish_points_clounds_without_detectObjs" value="true" />
    <param name="points_clounds_without_detectObjs" value="/points_clounds_without_detect_objects"/>
    <param name="cluster_merge_threshold" value="0.5" />
  </node>

</launch>
